src/os/HashIndex.cc

```diff
@@ -430,6 +431,7 @@ int HashIndex::pre_split_folder(uint32_t pg_num, uint64_t expected_num_objs)
     leavies = leavies >> 4;
   }
   for (uint32_t i = 0; i < subs; ++i) {
+    assert((!(split_bits % 4)) || split_bits <= 4); // otherwise BAD_SHIFT
```

Comment:
I think we just want to assert here that split_bits <= 4, based on the expression generating the number.